flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/utils/SerializerUtils.java

```diff
@@ -0,0 +1,98 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.mysql.source.utils;
+
+import org.apache.flink.core.memory.DataInputDeserializer;
+import org.apache.flink.core.memory.DataOutputSerializer;
+
+import com.ververica.cdc.connectors.mysql.source.offset.BinlogOffset;
+import com.ververica.cdc.connectors.mysql.source.offset.BinlogOffsetSerializer;
+import io.debezium.DebeziumException;
+import io.debezium.util.HexConverter;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+/** Utils for serialization and deserialization. */
+public class SerializerUtils {
+
+    private SerializerUtils() {}
+
+    public static void writeBinlogPosition(BinlogOffset offset, DataOutputSerializer out)
+            throws IOException {
+        out.writeBoolean(offset != null);
+        if (offset != null) {
+            byte[] binlogOffsetBytes = BinlogOffsetSerializer.INSTANCE.serialize(offset);
+            out.writeInt(binlogOffsetBytes.length);
+            out.write(binlogOffsetBytes);
+        }
+    }
+
+    public static BinlogOffset readBinlogPosition(int offsetVersion, DataInputDeserializer in)
+            throws IOException {
+        switch (offsetVersion) {
+            case 1:
+                return in.readBoolean() ? new BinlogOffset(in.readUTF(), in.readLong()) : null;
+            case 2:
+            case 3:
+                return readBinlogPosition(in);
+            default:
+                throw new IOException("Unknown version: " + offsetVersion);
+        }
+    }
+
+    public static BinlogOffset readBinlogPosition(DataInputDeserializer in) throws IOException {
+        boolean offsetNonNull = in.readBoolean();
+        if (offsetNonNull) {
+            int binlogOffsetBytesLength = in.readInt();
+            byte[] binlogOffsetBytes = new byte[binlogOffsetBytesLength];
+            in.readFully(binlogOffsetBytes);
+            return BinlogOffsetSerializer.INSTANCE.deserialize(binlogOffsetBytes);
+        } else {
+            return null;
+        }
+    }
+
+    public static String rowToSerializedString(Object[] splitBoundary) {
+        try (final ByteArrayOutputStream bos = new ByteArrayOutputStream();
+                ObjectOutputStream oos = new ObjectOutputStream(bos)) {
+            oos.writeObject(splitBoundary);
+            return HexConverter.convertToHexString(bos.toByteArray());
+        } catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)